src/v2/repositories/implementations/XcmRepository.ts

```diff
@@ -182,6 +182,68 @@ export class XcmRepository implements IXcmRepository {
     from: XcmChain,
     recipientAddress: string,
     amount: BN
+  ): Promise<ExtrinsicPayload> {
+    const isAstar = from.name === 'Astar';
```

Comment:
Is it possible to move this logic somewhere else (e.g. AstarXcmRepository). Since `XcmRepository` is base class it should be parachain agnostic.